### DISCUSSION TITLE
CheckConditionalAction LogicalOperator="And" only checks the first conditional

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14305] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

In events, using `<CheckConditionalAction targettag="X" LogicalOperator="And" >`  only picks the first of the conditionals listed.

Example:

This event spawn a clown, and check if the clown is both dead AND alive (two contradictory conditionals), which means the check should never succeed

In event 1, IsDead="true" is the first conditional
In event 2, IsDead="false" is the first conditional

Event 1 doesnt trigger until the clown is dead
Event 2 triggers immediately (clown is alive)

```xml
    <ScriptedEvent identifier="logicaloperatorbug1">
      <TagAction criteria="player" tag="player" />
      <SpawnAction npcsetidentifier="outpostnpcs1" npcidentifier="clown" targettag="clown" SpawnPointTag="player"/>
      <Label name="loop" />
      <CheckConditionalAction targettag="clown" LogicalOperator="And" > 
          <Conditional IsDead="True" />
          <Conditional IsDead="False" />
        <Failure>
          <WaitAction time="1" />
          <Goto name="loop" /> 
        </Failure>
        <Success>
          <ConversationAction text="Event 1 success" dialogtype="Small" targettag="player" endconversation="true" continueautomatically="true" />
        </Success>
      </CheckConditionalAction>
    </ScriptedEvent>
    
    <ScriptedEvent identifier="logicaloperatorbug2">
      <TagAction criteria="player" tag="player" />
      <SpawnAction npcsetidentifier="outpostnpcs1" npcidentifier="clown" targettag="clown" SpawnPointTag="player"/>
      <Label name="loop" />
      <CheckConditionalAction targettag="clown" LogicalOperator="And" > 
          <Conditional IsDead="False" />
          <Conditional IsDead="True" />
        <Failure>
          <WaitAction time="1" />
          <Goto name="loop" /> 
        </Failure>
        <Success>
          <ConversationAction text="Event 2 success" dialogtype="Small" targettag="player" endconversation="true" continueautomatically="true" />
        </Success>
      </CheckConditionalAction>
    </ScriptedEvent>
```

### Reproduction steps

1. Load a sub
2. triggerevent logicaloperatorbug1
3. Nothing
4. triggerevent logicaloperatorbug2
5. See text pop up
6. Kill 1st clown
7. Text pops up

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.6.101.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by dakkhuza** [comment]
Happens because of an [erroneously placed break statement](https://github.com/FakeFishGames/Barotrauma/blob/0aee53976202c7ea5c0bb3f26669a97c8dc2400c/Barotrauma/BarotraumaShared/SharedSource/Events/EventActions/CheckConditionalAction.cs#L48) in the loop that adds all conditionals to the internal list, the break causes the loop to always end after one conditional gets added, even if there are more conditionals to go through

**Comment by NilanthAnimosus** [comment]
Closing as fixed, should be included in a future patch (And has been merged with the code base).
